File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
    # 반절 만큼 구하고, 거기에서 텔레포트 한번 뛰면 최소다.
    
    jump = 0
    
    while n > 0:
        jump += n % 2
        n = n//2
        
    return jump

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int ans = 0;

        while(n > 0){
            if(n == 1){
                ans++;
                n--;
                continue;
            }
            
            // 순간이동이 가능하면
            if(n%2 == 0){
                n = n/2;
            }else{
                ans++;
                n--;
            }
        }

        return ans;
    }
}